uiowa/uiowa
# PHP 8.1 Upgrade

Relates to https://github.com/uiowa/uiowa/issues/5315
Resolves https://github.com/uiowa/uiowa/issues/5708

# To do

<details>
<summary>History</summary>

- [x] profiles/custom/sitenow @joewhitsitt
- [x] themes/custom/uids_base @pyrello
- [x] modules/custom/engineering_migrate
- [x] modules/custom/grad_custom @pyrello 
- [x] modules/custom/layout_builder_custom @pyrello 
- [x] modules/custom/sitenow_articles @joewhitsitt
- [x] modules/custom/sitenow_dcv @joewhitsitt - _no changes_
- [x] modules/custom/sitenow_dispatch @joewhitsitt
- [x] modules/custom/sitenow_events @joewhitsitt
- [x] modules/custom/sitenow_find_text @joewhitsitt - _no changes_
- [x] modules/custom/sitenow_intranet @joewhitsitt - _no changes_
- [x] modules/custom/sitenow_media_wysiwyg @joewhitsitt  
- [x] modules/custom/sitenow_migrate @joewhitsitt
- [x] modules/custom/sitenow_p2lb @joewhitsitt
- [x] modules/custom/sitenow_pages
- [x] modules/custom/sitenow_paragraphs @joewhitsitt
- [x] modules/custom/sitenow_people @joewhitsitt
- [x] modules/custom/sitenow_seo @joewhitsitt - _no changes_
- [x] modules/custom/studentlife_migrate @joewhitsitt - _no changes_
- [x] modules/custom/studentlife_topics @joewhitsitt - _no changes_
- [x] modules/custom/uiowa_aggregator @joewhitsitt
- [x] modules/custom/uiowa_alerts @joewhitsitt
- [x] modules/custom/uiowa_apr @joewhitsitt  - _no changes_
- [x] modules/custom/uiowa_area_of_study @joewhitsitt
- [x] modules/custom/uiowa_book @joewhitsitt - _no changes_
- [x] modules/custom/uiowa_core @joewhitsitt
- [x] modules/custom/uiowa_entities @joewhitsitt 
- [x] modules/custom/uiowa_graph @joewhitsitt 
- [x] modules/custom/uiowa_hours @joewhitsitt 
- [x] modules/custom/uiowa_maui @joewhitsitt 
- [x] modules/custom/uiowa_profiles @joewhitsitt 
- [x] modules/custom/uiowa_search @joewhitsitt - _no changes_
- [x] modules/custom/writinguniversity_migrate @joewhitsitt - _no changes_
- [x] modules/uiowa/uiowa_auth @joewhitsitt - https://github.com/uiowa/uiowa_auth/pull/6
- [x] sites/admissions.uiowa.edu @joewhitsitt 
- [x] sites/brand.uiowa.edu @joewhitsitt 
- [x] sites/coronavirus.uiowa.edu @joewhitsitt 
- [x] sites/dentistry.uiowa.edu @joewhitsitt 
- [x] sites/grad.admissions.uiowa.edu @joewhitsitt 
- [x] sites/grad.uiowa.edu @joewhitsitt 
- [x] sites/housing.uiowa.edu @joewhitsitt - _no changes_
- [x] sites/hr.uiowa.edu @joewhitsitt 
- [x] sites/ighn.international.uiowa.edu @joewhitsitt 
- [x] sites/iisc.uiowa.edu @joewhitsitt 
- [x] sites/neuroscience.grad.uiowa.edu @joewhitsitt - _no changes_
- [x] sites/now.uiowa.edu @joewhitsitt 
- [x] sites/pharmacy.uiowa.edu @joewhitsitt 
- [x] sites/sitenow.uiowa.edu - @GaryRidgway 
- [x] sites/sppa.uiowa.edu - @GaryRidgway 
- [x] sites/tippie.uiowa.edu - @GaryRidgway 
- [x] sites/transportation.uiowa.edu - @GaryRidgway 
- [x] sites/uiowa.edu - @GaryRidgway 
- [x] sites/uipress.uiowa.edu - @GaryRidgway 

Please track any migration comparison operators of old migrations here:
https://github.com/uiowa/uiowa/issues/5708

</details>

## Setup

It is recommended that your Docker and DDEV are updated:

- For docker, click the menu bar icon to download and install an update.
- `brew upgrade ddev` if you installed via homebrew

## To Test

- pull
- `ddev restart`
- `ddev composer install`
- `ddev auth ssh`
- `ddev ssh`
- `php -v` and confirm 8.1 or greater

## More Testing
- Generally take a look through this guide to see the types of things that are changing: https://www.php.net/manual/en/migration80.php
- Review the file changes to see if anything sticks out to you: https://github.com/uiowa/uiowa/pull/5611/files
- Review menu block as it relates to https://github.com/uiowa/uiowa/pull/5611/files#diff-22455fd1cfa159b58a0289784a91064192c27738f06b23e252be3a46271afb4cR609
- Review replicate as it relates to https://github.com/uiowa/uiowa/pull/5611/files#diff-96158dc2654c9de7f8206491f115c59eec3097e39643d343a870e538e89e5d6fR190
- Review v2/v3 functionality as it relates to the display title, hide headline, more link, use pager, link directly to source boolean options.
- Review that things previously restricted to Admins only are still working (they all use the same access check) so check to see that the advanced fontawesome icon options aren't visible to webmasters on event/timeline block.
- Make sure MAUI API is still sorting dates correctly based on timestamp.
- Check that there are no php errors while poking around law.dev, grad.dev or admissions.dev.

Semi-related, homebrew users should `brew install php@8.1` on their machines and switch to it at some point.
Afterward, update your PHPStorm > Preferences > PHP > CLI Interpreter
